gdx/src/com/badlogic/gdx/math/Matrix4.java

```diff
@@ -94,6 +95,56 @@ public Matrix4 () {
 		val[M33] = 1f;
 	}
 
+	private Quaternion getQuat () {
+		if (quat == null) quat = new Quaternion();
```

Comment:
Have you benchmark this? Checking for null in every call might slow the applications down. Have you noticed any difference?

Reply:
lazy instantiation has a very very very low impact in general. We did extensive benchmarks outlining that the jvm is doing a very good job here. Anyway i just modded the class so that only getTmpMat() is lazy. The rest is instance fields.